knowledge-base/adr/0020-generator-options.md

```diff
@@ -0,0 +1,60 @@
+
+## Decision 2
+
+| Current Name, Aliases                                    | Future Name, Aliases             | Current Behavior                                                                                  | Future Behavior                                                                                                                                              | Allow-no |
+| :------------------------------------------------------- | :------------------------------- | :------------------------------------------------------------------------------------------------ | :----------------------------------------------------------------------------------------------------------------------------------------------------------- | :------: |
+| `--inputDir`, `-i` _(required)_                          | `--input`, `-i` _(required)_     | Input directory/file                                                                              | _same_                                                                                                                                                       |  false   |
+| `--outputDir`, `-o` _(required)_                         | `--outputDir`, `-o` _(required)_ | Output directory                                                                                  | _same_                                                                                                                                                       |  false   |
+| `--clearOutputDir`                                       | `--clearOutputDir`               | Delete all files in output directory                                                              | _same_                                                                                                                                                       |  false   |
+| `--generateJs`                                           | `--transpile`, `-t`              | Transpiles, default true                                                                          | Transpiles, default false. Can only be set if tsconfig is enabled. This should be explicitly stated in the documentation.                                    |  false   |
+| `--generatePackageJson`                                  | `--packageJson`                  | Writes a default package.json, default true.                                                      | Writes a default package.json, default true (needs core dependency). Optionally in the future: Writes a custom package.json if passed. Keep boolean for now. |   true   |
+| `--serviceMapping`                                       | `--serviceMapping`               | Considers a service mapping file. Shows a warning if none is given (differs from OData behavior). | Considers a service mapping file. Generates it by default. Maybe in the future: Skips if set to false.                                                       |   true   |
```

Comment:
@jjtang1985 @FrankEssenberger Let me know, if we need additional alignment on this or whether here is good enough.

Reply:
In general I am not a big fan of the hybrid boolean or path options. I would say generate per default and if given consider the path.

Reply:
Updated it, is it better now?

Reply:
yes